golog.go

```diff
@@ -43,6 +43,9 @@ var (
 	bufferPool = bpool.NewBufferPool(200)
 
 	onFatal atomic.Value
+
+	// Impl by default uses the zap logger.
+	Impl = zapLogger
```

Comment:
Seems kind of dangerous to make this public without protection of a mutex or something like that. Would be safer to have a GetInstance() and SetInstance() that handle that. And in practice, we probably just need a SetInstance() anyway, right?

Reply:
Yeah makes sense. I’ll probably just create different constructors for different types or something.

Reply:
> I’ll probably just create different constructors for different types or something.

I like that.

Reply:
Now that I think of it, I think we need a way to late bind the logger implementation, since the API that clients mostly use is `LoggerFor` which shouldn't have to make any decisions about what's used under the covers. So, constructors wouldn't help here.